### PR TITLE
Wrap --help output to 120 characters

### LINKED ISSUE #7428: Using a password file
> I'm running:
>
> - **Which Parity version?**: 1.8.5b
> - **Which operating system?**  MacOS
> - **How installed?**: homebrew

### Issue
Unable to unlock parity with a password file.

```
~/G/tmp % parity --light --chain=ropsten \
          --unlock 0xaaf3ffee9d4c976aa8d0cb1bb84c3c90ee6e9118 \
          --password /Users/user/Github/tmp/unlock.txt \
          --bootnodes "enode://20c9ad97c081d63397d7b685a412227a40e23c8bdc6688c6f37e97cfbc22d2b4d1db1510d8f61e6a8866ad7f0e17c02b14182d37ea7c3c8b9c2683aeb6b733a
1@52.169.14.227:30303,enode://6ce05930c72abc632c58e2e4324f7c7ea478cec0ed4fa2528982cf34483094e9cbc9216e7aa349691242576d552a2a56aaeae426c5303ded677ce455ba1acd9d
@13.84.180.240:30303" \
          --jsonrpc-apis=all --ui-port 8080 ui
2018-01-02 14:22:49  Starting Parity/v1.8.5-beta-54bae9a-20171228/x86_64-macos/rustc1.22.1
2018-01-02 14:22:49  Keys path /Users/user/Library/Application Support/io.parity.ethereum/keys/Ropsten
2018-01-02 14:22:49  DB path /Users/user/Library/Application Support/io.parity.ethereum/chains/test/db/ae90623718e47d66
2018-01-02 14:22:49  Path to dapps /Users/user/Library/Application Support/io.parity.ethereum/dapps
2018-01-02 14:22:49  Running in experimental Light Client mode.
--bootnodes Unable to read password file. Ensure it exists and permissions are correct.
~/G/tmp % cat /Users/user/Github/tmp/unlock.txt
secret_password_here
~/G/tmp %
```

### PR DESCRIPTION
**Pull request by axelchalon**
The output of --help will now wrap at 100 characters (or to the terminal size if it's smaller than 100 characters). I used [textwrap](https://github.com/mgeisler/textwrap) and [term_size](https://github.com/kbknapp/term_size-rs), which are two libraries also used by Clap.

*textwrap* doesn't support correctly wrapping strings with line breaks, so for now I removed the line breaks in the few help messages that had some, and made a [PR](https://github.com/mgeisler/textwrap/pull/123) on textwrap to add support.

Also Clap fixed some bugs on their latest version (dependency version was bumped yesterday by Marek) and added support for global arguments now natively propagating up and down (so that they can be placed anywhere in the command), and so I removed the workaround I had used. The code is a bit simpler now.

Output of `parity --help`: https://gist.github.com/axelchalon/d2aa66d26e19ea8f5484150af1a2ea62
Output of `parity export state --help`: https://gist.github.com/axelchalon/69d50f8406ede0354176b94c8939a192

---

Closes #7428

### CODE REVIEW

**Comment by axelchalon** [comment]
Last commit closes https://github.com/paritytech/parity/pull/7626 and denies multiple values from being separated by spaces any longer, as this conflicts with values being able to start with hyphens (` --allow-ips -10.0.0.0/8`; `--password file1 file2 --other-option` used to be interpreted as `--password` with three values `file1` `file2` `--other-option`). Multiple values now need to be separated by a comma: `--password file1,file2 --other-option`.

File: parity/cli/usage.rs [hunk 1]
```diff
@@ -404,22 +419,27 @@ macro_rules! usage {
...
+
 				// Arguments and flags
 				$(
-					help.push_str("\n");
 					help.push_str($group_name); help.push_str(":\n");
 
 					$(
-						help.push_str(&format!("\t{}\n\t\t{}\n", $flag_usage, $flag_help));
+						help.push_str(&format!("    {}\n{}\n", $flag_usage,
+							indent(fill($flag_help, term_width-8).as_ref(),"        ")
```
Comment (andresilva) [inline parity/cli/usage.rs]: Could you extract this into a variable (const INDENT_PREFIX: &str = "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;";). And then you can replace the `8` with `INDENT_PREFIX.len()`.

**Comment by axelchalon** [comment]
@andresilva I think because it's not very readable to have such long lines on wide screens https://github.com/paritytech/parity/issues/2913#issuecomment-357685135

**Comment by 5chdn** [comment]
We can still use the terminal width, but limit it to 120 chars max.

**Comment by andresilva** [comment]
@axelchalon makes sense. I don't really have strong opinions on what the maximum should be (100 vs 120), maybe whatever looks better on wide screens.

https://gist.github.com/axelchalon/d2aa66d26e19ea8f5484150af1a2ea62#file-gistfile1-txt-L16
Maybe this should be indented? Not sure if it's easy to do it or not.

Overall the code and the resulting help output LGTM.

**Comment by axelchalon** [comment]
@5chdn currently, terminal width is used but limited to 100 characters. I don't have an opinion on 100 vs 120 characters either; whichever is preferred!

@andresilva yes, you're right. last commit fixes this; I updated the gists

**Comment by 5chdn** [comment]
Just played around with my terminal, 120 is pretty readable on a wide-screen. I would not go above that, and smaller terminals should just auto-detect the width. With high-resolution screens available I think it's fine to go for 120 over 100. 100 or even 80 (IBM punch card style) might be too narrow.

**Comment by axelchalon** [comment]
:+1: Gists are updated

**Comment by tomusdrw** [comment]
Why `B0-patch`? Does it fix any bugs?

**Comment by 5chdn** [comment]
@tomusdrw no I wanted this in for 1.9 but we missed that train. nvm.

**Pull request merged**